feat(perps): orderbook via feature flag

abretonc7s opened the pull request:
## **Description**

Re-enables the order book entry point from the market details view, now controlled by a LaunchDarkly feature flag for gradual rollout.

**Feature Flag Support:**
- Order Book visibility is controlled by LaunchDarkly flag `perps-order-book-enabled`
- Disabled by default - requires remote flag or `MM_PERPS_ORDER_BOOK_ENABLED="true"` env var to enable
- Supports version-gating via `minimumVersion` field in the flag configuration

**Documentation:**
- Added comprehensive Perps Feature Flags documentation at `docs/perps/perps-feature-flags.md`
- Covers architecture, implementation guide, LaunchDarkly configuration, and troubleshooting

The order book feature allows users to navigate from the market statistics card to view the full order book depth for a given market.

## **Changelog**

CHANGELOG entry: Re-enabled order book access from perps market details view

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: Order Book Navigation from Market Details

  Scenario: Order Book is hidden when feature flag is disabled
    Given MM_PERPS_ORDER_BOOK_ENABLED is not set or set to "false"
    And no LaunchDarkly remote flag is configured
    When the user opens the Perps market details view
    Then the "Order Book" row should NOT be visible in the statistics card

  Scenario: Order Book is visible when feature flag is enabled via env var
    Given MM_PERPS_ORDER_BOOK_ENABLED is set to "true"
    When the user opens the Perps market details view
    Then the "Order Book" row should be visible in the statistics card

  Scenario: User navigates to order book from market details
    Given the Order Book feature flag is enabled
    And the user is on the Perps market details view for any market
    When user taps the "Order Book" row in the statistics card
    Then the user is navigated to the Order Book view for that market
    And the order book displays bid/ask depth and spread information
```

## **Screenshots/Recordings**

### **Before**

<!-- Order book row was not visible in the statistics card -->

### **After**

<!-- Order book row is now visible and tappable in the statistics card -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Re-enables Order Book entry in Perps Market Details, controlled by a version-gated LaunchDarkly flag with env fallback, plus tests, mocks, and docs.
> 
> - **Perps UI**:
>   - Re-enable Order Book navigation in `PerpsMarketDetailsView` and show clickable row in `PerpsMarketStatisticsCard` when `onOrderBookPress` is provided.
> - **Feature Flags**:
>   - Add selector `selectPerpsOrderBookEnabledFlag` with version-gated remote flag (`perpsOrderBookEnabled`) and local fallback `MM_PERPS_ORDER_BOOK_ENABLED`.
>   - Wire selector into `PerpsMarketDetailsView` to conditionally expose Order Book entry.
>   - Update mocks to include `perpsOrderBookEnabled`.
> - **Tests**:
>   - Add comprehensive unit tests for `selectPerpsOrderBookEnabledFlag` covering defaults, remote precedence, version gating, and fallbacks.
> - **Docs**:
>   - Add `docs/perps/perps-feature-flags.md` detailing the Perps feature-flag framework and usage.
> - **Config**:
>   - Add `MM_PERPS_ORDER_BOOK_ENABLED` to `.js.env.example`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e3291291cfbae14ddb0260cb99fa3f31ba27535a. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

abretonc7s committed:
feat(perps): re-enable order book

gambinish reviewed (changes requested):
Can we please create a feature flag for this to give us flexibility in our release? We discussed this briefly during our offsite. This way can confidently merge, and enable it on experimental while keeping it hidden behind a feature flag on production.

I think that we will need to get some design feedback on the order book, since this was built without designs

abretonc7s committed:
feat: add feature flag to orderbook

abretonc7s committed:
refactor: update Perps feature flag behavior to be disabled by default

- Changed the default behavior of the `perpsOrderBookEnabled` feature flag to be disabled by default, updating related tests and documentation accordingly.
- Adjusted the logic in `selectPerpsOrderBookEnabledFlag` to reflect this change, ensuring that the flag returns false when no remote or local environment variable is set.
- Updated test cases to verify the new default behavior and modified documentation to reflect the updated default state of the feature flag.

abretonc7s committed:
Merge branch 'main' into feat/perps/order-book-again

gambinish reviewed (dismissed):
Approving! Thanks for feature flagging

abretonc7s committed:
Merge branch 'main' into feat/perps/order-book-again

abretonc7s commented on .js.env.example:
```diff
@@ -161,6 +161,7 @@ export MM_PERPS_ENABLED="true"
 export MM_PERPS_SERVICE_INTERRUPTION_BANNER_ENABLED="false"
 export MM_PERPS_BLOCKED_REGIONS="US,CA-ON,GB,BE"
 export MM_PERPS_GTM_MODAL_ENABLED="true"
+export MM_PERPS_ORDER_BOOK_ENABLED="true"
```
it is disabled by default, setting to true enables it

abretonc7s committed:
feat: simplify docs

abretonc7s committed:
Merge branch 'main' into feat/perps/order-book-again
